### ISSUE TITLE
Option to disable MAC address lookup

### ISSUE THREAD

**Issue by qopisbel** [aaronjwood/PortAuthority#11]
Could occur at first app start (opt-in) and/or in a settings menu (opt-out). Some users may not want MAC info automatically shared invisibly with a third party.

**Comment by ocdtrekkie** [comment]
++

**Comment by aaronjwood** [comment]
Good idea, I'll implement this.

**Comment by aaronjwood** [comment]
Or maybe instead I'll bring in a database locally and stop calling out to an api

**Comment by aaronjwood** [comment]
I have decided to move away from using this service altogether due to the reasons here #12

**Comment by qopisbel** [comment]
Thanks, that seems like the best option (although it is more work). :+1:

**Comment by aaronjwood** [comment]
You're welcome. The work will be well worth it :) I've just started so hopefully in a few days I'll be ready to release another update with this service removed.

**Comment by qopisbel** [comment]
I wonder if there are any individuals or groups that use the traffic from those MAC lookup services to map internal networks? There are probably easier ways to get that data, and it may be of varying usefulness, but given extent of passive surveillance on the Internet nothing would really surprise me anymore.

**Comment by ocdtrekkie** [comment]
Awesome choice, sir. I prefer apps with local smarts where possible, and the ability to disable cloud or remote services when not possible. Always great to find security-minded devs.

**Comment by aaronjwood** [comment]
:D I always try to be very security conscious. I think I made a pretty poor call by choosing an API for MAC vendor info now that I look back on it.

**Comment by aaronjwood** [comment]
Well, the database stuff was a bit quicker to implement than I expected :) I've pushed all related changes into the feature/mac-vendor-sqlite branch, tested them, and merged that branch into the development branch. If anyone here wants to build the APK from source and try it out I'd greatly appreciate it. There's no auto update functionality around the database (for now anyways) but I plan to come back to that in the near future. For now I've made a database from the very latest version of http://standards.ieee.org/regauth/oui/oui.txt
